bindings/nodejs/src/structures/sol-js-gpio.cc

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <nan.h>
+#include "sol-js-gpio.h"
+#include "../common.h"
+#include "handles.h"
+
+using namespace v8;
+
+static sol_gpio_edge sol_edge_from_string(Local<Value> jsString) {
```

Comment:
as I said this would be better done in `sol-gpio.h`, that is in the C API. You can also do the other way around, like enum->string. In this sense, for the string->enum, you can use the `sol_str_table` struct. For the enum->string, you can use a simple array with `[value] = string` and validate `if (value >= sizeof(array)) return error;`

Examples:
 - https://github.com/solettaproject/soletta/blob/master/src/lib/common/sol-log.c#L95 string->enum;
 - https://github.com/solettaproject/soletta/blob/master/src/lib/common/sol-log.c#L357 enum->string;

you can name all these functions in the pattern `const char *<enum_name>_to_str(enum)` and `enum <enum_name>_from_str(const char *)`, like `sol_gpio_edge_to_str()` and `sol_gpio_edge_from_str()`